### ISSUE TITLE
Does MigrationExecutor or runMigrations() need a QueryRunner.release()?

### ISSUE THREAD

**Issue by tma-isbx** [typeorm/typeorm#1201]
Scenarios 1:
If a script calls connection.runMigrations(), the script will not terminate as the connection remains open. The CLI solves this with an explicit connection.close(). This may still be desirable to avoid having to wait for database connection timeouts.

Scenario 2:
1. DB's connection pool is configured to 1 connection.
2. Call connection.runMigrations().
3. Attempt to call connection.query().
Expected result: the query is executed after the migrations.
Actual result: the query is blocked and never executes.

**Comment by pleerock** [comment]
Yes it must call it in the case when it creates query runner on his own, so following code:
```ts
    constructor(protected connection: Connection, queryRunner?: QueryRunner) {
        this.queryRunner = queryRunner || connection.createQueryRunner("master");
    }
```

is not correct in `MigrationExecutor`

**Comment by pleerock** [comment]
This is fixed and released in `0.1.5`, can you please check it?

**Comment by tma-isbx** [comment]
Looks good, it has resolved my issue. Thanks!
